### PR TITLE
Fix warnings, fix wrong units in benchmark

### PR DESCRIPTION
**Pull request by janciesko**

### CODE REVIEW

File: benchmarks/access_overhead/access_overhead_teams.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ struct Access<ViewType_t, typename std::enable_if_t<!std::is_same<
                                ? first_i + iters_per_team + iters_per_team_mod
                                : first_i + iters_per_team;
     Kokkos::parallel_for(Kokkos::TeamThreadRange(thread, first_i, last_i),
-                         [=](const StreamIndex i) { v(i) += 1; });
+                         [=, this](const StreamIndex i) { v(i) += 1; });
```
Comment (dalg24) [inline benchmarks/access_overhead/access_overhead_teams.cpp:133]: That does not look right.

**Review by dalg24 — COMMENTED** [review]
You are mixing all sorts of unrelated changes which makes it harder to review and audit later.

File: benchmarks/access_overhead/access_overhead_teams.cpp (hunk 1 above)
Comment (janciesko) [reply benchmarks/access_overhead/access_overhead_teams.cpp:133]: What in particular?

Comment (dalg24) [reply benchmarks/access_overhead/access_overhead_teams.cpp]: why did you add it?

Comment (janciesko) [reply benchmarks/access_overhead/access_overhead_teams.cpp]: C++20 requires explicit capture of `this`. Note that this is a inner-nested `parallel_for` and capturing by value results in capturing a valid `this` ptr.

**Review by devreal — APPROVED** [review]

**Pull request merged**